qa_tool/tests/test_relevancy_score.py

```
@@ -0,0 +1,255 @@
+        url = '/qa/v1/relevancyscores/{0}/'.format(score.pk)
+        self.data['value'] = an_integer()
+        response = self.APIClient.put(url, self.data)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class POSTRelevancyScoreTests(TestCase):
+    def setUp(self):
+        self.user = self.make_user()
+        self.token = Token.objects.create(user=self.user)
+        self.APIClient = APIClient()
+        self.score_value = an_integer()
+        self.data_without_algorithm_in_body = {
+            'value': self.score_value,
+            'search_location': SearchLocationBuilder().create().id,
+            'service_at_location': ServiceAtLocationBuilder().create().id,
+            'topic': create_topic('test').id,
+        }
+        self.data_with_algorithm_in_body = {
+            'value': self.score_value,
+            'algorithm': AlgorithmBuilder().create().id,
+            'search_location': SearchLocationBuilder().create().id,
+            'service_at_location': ServiceAtLocationBuilder().create().id,
+            'topic': create_topic(a_string()).id,
+        }
+
+    def test_cannot_post_unauthenticated(self):
+        url = '/qa/v1/relevancyscores/'
+        response = self.APIClient.post(url, data=self.data_with_algorithm_in_body)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_can_post_with_algorithm_in_body_not_in_url_short_url(self):
+        url = '/qa/v1/relevancyscores/'
+        self.APIClient.credentials(HTTP_AUTHORIZATION='Token ' + self.token.key)
+        response = self.APIClient.post(url, data=self.data_with_algorithm_in_body)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.json()['value'], self.score_value)
+
+    def test_cannot_post_with_algorithm_in_body_not_in_url_long_url(self):
+        url = '/qa/v1/algorithms/not_algorithm_id/relevancyscores/'
```

Test name is misleading, the algorithm id *is* in the URL, it's just an invalid one. This end point should return an error even with a valid id in the url and the body.

---

I have decided to change all `not_algorithm_id` or equivalent to be 0 to be consistent.

---

I have decided to include the old tests but rename them to be more appropriate